frontend/src/components/worksheets/WorksheetNameSearch.js

```diff
@@ -11,12 +11,20 @@ export default class extends React.Component {
         };
     }
     async componentDidMount() {
-        const { name } = queryString.parse(this.props.location.search);
+        let { name } = queryString.parse(this.props.location.search);
+        if (name === undefined && this.props.auth.isAuthenticated) {
```

Comment:
nit: Would it be better to check if the name is undefined once and nest the if-else statement that checks `isAuthenticated` or use a ternary operator?

Reply:
Yeah that makes sense! I will change this before merging